packages/api/test/unit/beacon/testData/validator.ts

```diff
@@ -33,23 +33,35 @@ export const testData: GenericServerTestCases<Api> = {
     args: [1000],
     res: {
       executionOptimistic: true,
-      data: [{slot: 1, validatorIndex: 2, pubkey: Buffer.alloc(48, 3)}],
+      data: [{slot: 1, validatorIndex: 2, pubkey: new Uint8Array(48).fill(3)}],
       dependentRoot: ZERO_HASH_HEX,
     },
   },
   getSyncCommitteeDuties: {
     args: [1000, [1, 2, 3]],
     res: {
       executionOptimistic: true,
-      data: [{pubkey: Buffer.alloc(48, 1), validatorIndex: 2, validatorSyncCommitteeIndices: [3]}],
+      data: [{pubkey: Uint8Array.from(Buffer.alloc(48, 1)), validatorIndex: 2, validatorSyncCommitteeIndices: [3]}],
     },
   },
   produceBlock: {
-    args: [32000, randaoReveal, graffiti],
+    args: [
+      32000,
+      randaoReveal,
+      graffiti,
+      undefined,
+      {feeRecipient: undefined, builderSelection: undefined, strictFeeRecipientCheck: undefined},
+    ],
```

Comment:
`sinon` assertions were passing with partial arguments as well. To pass with the `vitest` we have to add the full arguments that were used to call that function.

Reply:
because produceBlockV3 serializer is superset of all of them

Reply:
you can create separate serializer fns for each of these but then delegate the call to `produceBlockV3` inside it with so that you may not need to pass undefined values here

but we anyway will remove these fns and not support them so we can leave it like that as well